specs/ExtendedProcessFailed.md

```diff
@@ -0,0 +1,671 @@
+  /// is `COREWEBVIEW2_PROCESS_FAILED_KIND_BROWSER_PROCESS_EXITED`, and
+  /// `COREWEBVIEW2_PROCESS_FAILED_REASON_UNRESPONSIVE` when `ProcessFailedKind`
+  /// is `COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_UNRESPONSIVE`.
+  /// For other process failure kinds, the reason may be any of the reason
+  /// values.
+  [propget] HRESULT Reason(
+      [out, retval] COREWEBVIEW2_PROCESS_FAILED_REASON* reason);
+
+  /// The exit code of the failing process. The exit code is always `1` when
+  /// `ProcessFailedKind` is
+  /// `COREWEBVIEW2_PROCESS_FAILED_KIND_BROWSER_PROCESS_EXITED`, and
+  /// `STILL_ACTIVE` (`259`) when `ProcessFailedKind` is
+  /// `COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_UNRESPONSIVE`.
+  [propget] HRESULT ExitCode(
+      [out, retval] int* exitCode);
+
+  /// Description of the process assigned by the WebView2 Runtime. This is a
+  /// technical English term appropriate for logging or development purposes,
+  /// and not localized for the end user. It applies to utility processes (for example,
+  /// "Audio Service" or "Video Capture") and plugin processes (for example, "Flash").
+  /// The returned `processDescription` is `null` if the WebView2 Runtime did
+  /// not assign a description to the process.
+  [propget] HRESULT ProcessDescription(
+      [out, retval] LPWSTR* processDescription);
+
+  /// The list of frames in the `CoreWebView2` that were being rendered by the
+  /// failed process. The content in these frames is replaced with an error page.
+  /// This is only available when `ProcessFailedKind` is
+  /// `COREWEBVIEW2_PROCESS_FAILED_KIND_FRAME_RENDER_PROCESS_EXITED`;
+  /// `frames` is `null` for all other process failure kinds, including the case
+  /// in which the failed process was the renderer for the main frame and
+  /// subframes within it, for which the failure kind is
+  /// `COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_EXITED`.
+  [propget] HRESULT ImpactedFramesInfo(
+      [out, retval] ICoreWebView2FrameInfoCollection** frames);
+}
+
+/// Collection of frame details (name and source). Used to list the impacted
+/// frames' info when a frame-only render process failure occurs in the
+/// `ICoreWebView2`.
+[uuid(4bedeef8-3de7-4a3a-aadc-e9437bfb3e92), object, pointer_default(unique)]
+interface ICoreWebView2FrameInfoCollection : IUnknown {
+  /// Gets an iterator over the collection of frames' info.
+  HRESULT GetIterator(
+      [out, retval] ICoreWebView2FrameInfoCollectionIterator** iterator);
+}
+
+/// Iterator for a collection of frames' info. For more info, see
+/// `ICoreWebView2ProcessFailedEventArgs2` and
+/// `ICoreWebView2FrameInfoCollection`.
+[uuid(0e2367b9-c725-4696-bb8a-75b97af32330), object, pointer_default(unique)]
+interface ICoreWebView2FrameInfoCollectionIterator : IUnknown {
+  /// Get the current `ICoreWebView2FrameInfo` of the iterator.
+  HRESULT GetCurrentFrameInfo([out, retval] ICoreWebView2FrameInfo** frameInfo);
```

Comment:
Is this the existing naming pattern for iterators? Usually, the iterator members are just named `Current`, `HasCurrent`, and `MoveNext()`.

Reply:
Removed.